sql/logictest: TestLogic failed

sql/logictest.TestLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3519417&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3519417&tab=artifacts#/) on master @ [f4a1848e2863d8aa04688caaa3520cd5cbe9a098](https://github.com/cockroachdb/cockroach/commits/f4a1848e2863d8aa04688caaa3520cd5cbe9a098):


```
=== RUN   TestLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic268683927
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestLogic
    logic.go:3437: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic268683927
--- FAIL: TestLogic (2659.13s)
=== RUN   TestLogic/experimental-span-configs
    --- FAIL: TestLogic/experimental-span-configs (0.00s)
=== RUN   TestLogic/experimental-span-configs/auto_span_config_reconciliation_job
=== PAUSE TestLogic/experimental-span-configs/auto_span_config_reconciliation_job
=== CONT  TestLogic/experimental-span-configs/auto_span_config_reconciliation_job
    logic.go:2323: 
         
        testdata/logic_test/auto_span_config_reconciliation_job:8: SELECT count(*), job_type, status FROM [SHOW AUTOMATIC JOBS] WHERE job_type = 'AUTO SPAN CONFIG RECONCILIATION' GROUP BY (job_type, status)
        expected:
            count  job_type                         status
            1      AUTO SPAN CONFIG RECONCILIATION  running
            
        but found (query options: "colnames") :
            count  job_type  status
--- done: testdata/logic_test/auto_span_config_reconciliation_job with config experimental-span-configs: 2 tests, 1 failures
    logic.go:3205: 
        testdata/logic_test/auto_span_config_reconciliation_job:14: error while processing
    logic.go:3205: testdata/logic_test/auto_span_config_reconciliation_job:14: too many errors encountered, skipping the rest of the input
--- total: 152073 tests, 0 failures
        --- FAIL: TestLogic/experimental-span-configs/auto_span_config_reconciliation_job (0.35s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=2
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63262 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-20.2]
- #61657 sql/logictest: TestLogic failed [C-test-failure O-robot branch-release-21.1]
- #58269 sql/logictest: TestLogic failed [C-test-failure O-robot T-sql-schema branch-release-20.1]
</p>
</details>
/cc @cockroachdb/sql-queries jordanlewis
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/auto_span_config_reconciliation_job
+++ b/pkg/sql/logictest/testdata/logic_test/auto_span_config_reconciliation_job
@@ -5,7 +5,7 @@ SET CLUSTER SETTING spanconfig.experimental_reconciliation_job.enabled = true;
 
 # Ensure there's a single auto span config reconciliation job in the system,
 # and that it's running.
-query ITT colnames
+query ITT colnames,retry
 SELECT count(*), job_type, status FROM [SHOW AUTOMATIC JOBS] WHERE job_type = 'AUTO SPAN CONFIG RECONCILIATION' GROUP BY (job_type, status)
 ----
 count  job_type                         status

